selene/src/main.rs

```diff
@@ -391,6 +390,58 @@ fn read_file(checker: &Checker<toml::value::Value>, filename: &Path) {
     );
 }
 
+/// Reads a config file and return content and path. None if not found or empty.
+
+fn read_config_file() -> (String, Option<PathBuf>) {
+    const CONFIG_PATHS_TO_CHECK: [&str; 2] = [".selene.toml", "selene.toml"];
+
+    let (config_contents, config_path) =
+        match CONFIG_PATHS_TO_CHECK
+            .iter()
+            .find_map(|path| {
+                match fs::read(path) {
+                    Ok(contents) => {
+                        if let Ok(valid_str) = String::from_utf8(contents) {
+                            Some((valid_str, PathBuf::from(path)))
+                        } else {
+                            // Handle invalid UTF-8
+                            error!("Error reading config file: {}, stream did not contain valid UTF-8", path);
+                            std::process::exit(1);
+                        }
+                    }
+                    Err(_) => None, // if a file do not exist, return none

```

Comment:
@chriscerie thank you for being so explicit in your answer. It should be fixed in 
https://github.com/Kampfkarren/selene/pull/591/commits/aff6f9a7a0719df393badf4a82442b5f519771ce
https://github.com/Kampfkarren/selene/pull/591/commits/7c844a3d5a91b680a9d20c17665e7b6544ba2c64

Every manual test I've performed seem to work as expected. Including permission denied.
![screenshot_2024-05-08_21-21-31_356340982](https://github.com/Kampfkarren/selene/assets/3357792/1c7268e6-289f-4213-9c8d-90280b640b4a)

Bad UTF-8
![screenshot_2024-05-08_22-13-15_154181796](https://github.com/Kampfkarren/selene/assets/3357792/d0b688e2-6018-4176-a971-089a374b2b73)

## About the missing config error
We probably don't want to display the `missing config` error, because:
* The error would also display when using selene normally (not only when doing `selene validate-config`).
* It's not really an error, as selene will fallback to the default config.
* The error would display in the code editor, which is something we prob don't want.

A workaround we could use is handling the error with `println!` instead of `error!` so it prints the error when using selene on the terminal, but not on code editors. I've tested it and it works, but for now I'm just gonna silence the error unless you tell me otherwise. If you decide to enable it would look like:

![screenshot_2024-05-08_22-22-58_734164313](https://github.com/Kampfkarren/selene/assets/3357792/d1ba249d-740d-4784-9a58-be9fb24316eb)

Reply:
> The error would also display when using selene normally (not only when doing selene validate-config).

The existing behavior for missing config with `validate-config` is to error which behaves differently from the main command, and that should not be changed. The two commands are handled separately. See https://github.com/Kampfkarren/selene/blob/5c02fe9b9e25137b5896dac5e941854353cdb200/selene/src/main.rs#L456-L457

> It's not really an error, as selene will fallback to the default config.

It's not intuitive to me for the command to fallback to the default config when its purpose is to validate the user's config.

> The error would display in the code editor, which is something we prob don't want.

Where would it display in the code editor?

Reply:
@chriscerie Ok the last commit should solve it:

```rust
let error = io::Error::new(
  io::ErrorKind::NotFound,
 "Error reading config file: No such file or directory (os error 2)"
);
error!("{}", error);
```

It displays on `./selene validate-config`
![screenshot_2024-05-21_15-35-42_777689703](https://github.com/Kampfkarren/selene/assets/3357792/e27eaff1-c1bd-448c-97fc-e91ffba13397)

All the border cases seem to work fine too:
* The error display when running when running validate-config if none of the files ./.selene or ./selene is present.
* The error doesn't display on the code editor, only on validate-config.
* The error doesn't display when running validate-config if ./.selene or ./selene exist are the file is correct.

Reply:
Ok I solved all border cases. Ready to review.